onboard-server/src/main/java/no/vegvesen/ixn/serviceprovider/OnboardRestController.java

```diff
@@ -300,59 +302,106 @@ public GetSubscriptionResponse getServiceProviderSubscription(@PathVariable Stri
+		PrivateChannel privateChannelToUpdate = privateChannelRepository.findByServiceProviderNameAndId(serviceProviderName, parsedId);
+		if (privateChannelToUpdate == null) {
+			throw new NotFoundException("The private channel to delete is not in the Service Provider private channels. Cannot delete private channel that don't exist.");
+		}
 
 		// Save updated Service Provider - set it to TEAR_DOWN. It's the routing-configurers job to delete from the database, if needed.
-		ServiceProvider saved = serviceProviderRepository.save(serviceProviderToUpdate);
-		logger.debug("Updated Service Provider: {}", saved.toString());
+		privateChannelToUpdate.setStatus(PrivateChannelStatus.TEAR_DOWN);
+		PrivateChannel saved = privateChannelRepository.save(privateChannelToUpdate);
 
-		RedirectView redirect = new RedirectView("/{serviceProviderName}/privatechannels/");
-		OnboardMDCUtil.removeLogVariables();
-		return redirect;
+		logger.debug("Updated Private Channel: {}", saved);
 
+		OnboardMDCUtil.removeLogVariables();
 	}
 
 	@RequestMapping(method = RequestMethod.GET, path = "/{serviceProviderName}/privatechannels", produces = MediaType.APPLICATION_JSON_VALUE)
-	public PrivateChannelListApi getPrivateChannels(@PathVariable String serviceProviderName) {
+	public ListPrivateChannelsResponse getPrivateChannels(@PathVariable String serviceProviderName) {
 		OnboardMDCUtil.setLogVariables(nodeProperties.getName(), serviceProviderName);
 		logger.info("listing private channels for service provider {}", serviceProviderName);
 		this.certService.checkIfCommonNameMatchesNameInApiObject(serviceProviderName);
 
-		ServiceProvider serviceProvider = getOrCreateServiceProvider(serviceProviderName);
+		List<PrivateChannel> privateChannels = privateChannelRepository.findAllByServiceProviderName(serviceProviderName);
+
+		OnboardMDCUtil.removeLogVariables();
+		return typeTransformer.transformPrivateChannelListToListPrivateChannels(serviceProviderName, privateChannels);
+	}
+
+	@RequestMapping(method = RequestMethod.GET, path = "/{serviceProviderName}/privatechannels/{privateChannelId}")
+	public GetPrivateChannelResponse getPrivateChannel(@PathVariable String serviceProviderName, @PathVariable String privateChannelId) {
+		OnboardMDCUtil.setLogVariables(nodeProperties.getName(), serviceProviderName);
+		logger.info("Get private channel {} for service provider {}", privateChannelId, serviceProviderName);
+		this.certService.checkIfCommonNameMatchesNameInApiObject(serviceProviderName);
 
-		List<PrivateChannel> privateChannels = serviceProvider.getPrivateChannels().stream().collect(Collectors.toList());
-		List<PrivateChannelApi> privateChannelsApis = new ArrayList<>();
-		for(PrivateChannel privateChannel : privateChannels){
-			privateChannelsApis.add(new PrivateChannelApi(privateChannel.getPeerName(), privateChannel.getQueueName(), privateChannel.getId()));
+		Integer parsedId;
+
+		try{
+			parsedId = Integer.parseInt(privateChannelId);
+		}catch (Exception e){
```

Comment:
Never catch Exception. Catch a NumberFormatException instead. This is due to Exception being all kinds of possible exceptions, and we only want to catch the one we expect to happen due to the parsing.